cmake/target/target.cmake

```diff
@@ -110,18 +132,28 @@ function(setup_single_target TARGET_FILE MODULE SOURCES DEPENDENCIES)
     if (CMAKE_DEBUG_OUTPUT)
         message(STATUS "[target] Setting up '${TARGET_NAME}' on all module ${MODULE}")
     endif()
-    get_target_property(MODULE_TYPE "${MODULE}" FP_TYPE)
+    get_target_property(MODULE_TYPE "${MODULE}" FPRIME_TYPE)
 
     if (NOT MODULE_TYPE STREQUAL "Deployment")
         cmake_language(CALL "${TARGET_NAME}_add_module_target" "${MODULE}" "${TARGET_NAME}" "${SOURCES}" "${DEPENDENCIES}")
     else()
-        get_target_property(RECURSIVE_DEPENDENCIES "${MODULE}" FP_RECURSIVE_DEPS)
-        if (NOT RECURSIVE_DEPENDENCIES)
-            resolve_dependencies(RESOLVED ${DEPENDENCIES})
-            recurse_targets("${MODULE}" RECURSIVE_DEPENDENCIES "" ${RESOLVED})
-            set_target_properties("${MODULE}" PROPERTIES FP_RECURSIVE_DEPS "${RECURSIVE_DEPENDENCIES}")
+        get_target_property(TRANSITIVE_LINK_LIBRARIES "${MODULE}" TRANSITIVE_LINK_LIBRARIES)
+        # Recalculate recursive dependencies
+        if (NOT TRANSITIVE_LINK_LIBRARIES)
+            recurse_target_property("${MODULE}" INTERFACE_LINK_LIBRARIES KNOWN_TRANSITIVE_LINKS EXTERNAL_LINKS UNKNOWN_LINKS)
+            # Report all detected recursive dependencies
+            if (CMAKE_DEBUG_OUTPUT)
+                foreach(LIST_PRINT IN ITEMS EXTERNAL_LINKS KNOWN_TRANSITIVE_LINKS UNKNOWN_LINKS)
+                    message(STATUS "'${MODULE}' Recursive Links: ${LIST_PRINT}")
+                    foreach(ITEM_PRINT IN LISTS ${LIST_PRINT})
+                        message(STATUS "    ${ITEM_PRINT}")
+                    endforeach()
+                endforeach()
+            endif()
+            check_unknown_links("${MODULE}" ${UNKNOWN_LINKS})
+            set_target_properties("${MODULE}" PROPERTIES TRANSITIVE_LINK_LIBRARIES "${KNOWN_TRANSITIVE_LINKS}")
         endif()
-        cmake_language(CALL "${TARGET_NAME}_add_deployment_target" "${MODULE}" "${TARGET_NAME}" "${SOURCES}" "${DEPENDENCIES}" "${RECURSIVE_DEPENDENCIES}")
+        cmake_language(CALL "${TARGET_NAME}_add_deployment_target" "${MODULE}" "${TARGET_NAME}" "${SOURCES}" "${DEPENDENCIES}" "${TRANSITIVE_LINK_LIBRARIES}")
```

Comment:
Update TRANSITIVE_LINK_LIBRARIES after it's set, or read it in _add_deployment_target